scripts/install_plugin_modules.js

```diff
@@ -0,0 +1,108 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const requireDir = require('require-dir')
+const crypto = require('crypto')
+const semver = require('semver')
+const exec = require('./helpers/exec')
+const plugins = requireDir('../src/plugins')
+
+const workspaces = new Set()
+
+run()
+
+function run () {
+  assertVersions()
+  assertWorkspace()
+  install()
+}
+
+function assertVersions () {
+  Object.keys(plugins).filter(key => key !== 'index').forEach(key => {
+    [].concat(plugins[key]).forEach(instrumentation => {
+      [].concat(instrumentation.versions).forEach(version => {
+        if (version) {
+          assertModules(instrumentation.name, version)
+          assertModules(instrumentation.name, semver.coerce(version).version)
+        }
+      })
+    })
+  })
+}
+
+function assertModules (name, version) {
+  addFolder(name, version)
+  assertFolder(name, version)
+  assertPackage(name, version)
+  assertIndex(name, version)
+}
+
+function assertFolder (name, version) {
+  if (!fs.existsSync(folder())) {
+    fs.mkdirSync(folder())
+  }
+
+  if (!fs.existsSync(folder(name, version))) {
+    fs.mkdirSync(folder(name, version))
+  }
+}
+
+function assertPackage (name, version) {
+  fs.writeFileSync(filename(name, version, 'package.json'), JSON.stringify({
+    name: [name, sha1(version)].filter(val => val).join('-'),
+    version: '1.0.0',
+    license: 'BSD-3-Clause',
+    private: true,
+    dependencies: {
+      [name]: version
+    }
+  }, null, 2) + '\n')
+}
+
+function assertIndex (name, version) {
+  const index = `'use strict'
+
+module.exports = {
+  get (id) { return require(id || '${name}') },
+  version () { return require('${name}/package.json').version }
+}
+`
+  fs.writeFileSync(filename(name, version, 'index.js'), index)
+}
+
+function assertWorkspace () {
+  fs.writeFileSync(filename(null, null, 'package.json'), JSON.stringify({
+    name: 'versions',
+    version: '1.0.0',
+    license: 'BSD-3-Clause',
+    private: true,
+    workspaces: Array.from(workspaces)
+  }, null, 2) + '\n')
+}
+
+function install () {
+  exec('yarn', { cwd: folder() })
```

Comment:
This doesn’t work on a fresh clone and using Node v4 as per the engine version specified in `package.json`, because the `mysql2` dev dependency requires Node >= v6. Maybe change this to `yarn --ignore-engines` ?

Reply:
The problem with `yarn --ignore-engines` is that since the module no longer supports Node 4, they probably are using features available in Node 6, meaning that the test would still fail (either now or in the future).

What I think could be done is to set plugin dependencies in `optionalDependencies`, and simply skip the test if the module is not available (which would mean the module was not installed for the current Node version).

Reply:
Last week I was able to run the tests fine on node 4, so I assumed that maybe the node version requirement was a bit of an overzealous one, but I can’t argue with your idea which seems the most future-proof 👍